ios/FluentUI/Vnext/AvatarGroup/AvatarGroup.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+@objc public enum MSFAvatarGroupStyle: Int, CaseIterable {
+    case stack
+    case pile
+}
+
+/// `MSFAvatarGroupState` defines properties in an AvatarGroup
+///
+/// `maxDisplayedAvatars`: Caps the number of displayed avatars and shows the remaining not displayed in the overflow avatar
+///
+/// `overflowCount`: Adds to the overflow count in case the calling code did not provide all the avatars, but still wants to convey more
+/// items than just the remainder of the avatars that could not be displayed due to the maxDisplayedAvatars property.
+///
+/// `style`: Type of AvatarGroup. Avatars in a Stack is layered on top of each other; Pile separates avatars
+///
+/// `createAvatar`: Creates an Avatar using an AvatarStyle and AvatarSize
+///
+/// `getAvatarState`: Gets the state of an Avatar using an index value
+///
+/// `deleteAvatar`: Deletes an Avatar with a specified index value
+@objc public protocol MSFAvatarGroupState {
+    var maxDisplayedAvatars: Int { get set }
+    var overflowCount: Int { get set }
+    var style: MSFAvatarGroupStyle { get set }
+
+    func createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize) -> MSFAvatarState?
+    func getAvatarState(at index: Int) -> MSFAvatarState
+    func deleteAvatar(at index: Int)
+}
+
+/// Properties that make up AvatarGroup content
+class MSFAvatarGroupStateImpl: NSObject, ObservableObject, MSFAvatarGroupState {
+    func createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize) -> MSFAvatarState? {
+        avatars.append(MSFAvatarStateImpl(style: style, size: size))
```

Comment:
It's safer and less error prone to store the return of the
`MSFAvatarStateImpl(style: style, size: size)`
in a constant, then add it to the array and return the constant you just added.

There is a potential risk for a race condition and multiple threads calling this at the same time could cause 2 different calls to return the same avatarstate instance.